### PR TITLE
Break down SCIM modules and annotate SCIM tests

### PR DESCRIPTION
**Pull request by neongreen**
No new tests yet but there will be some.

### CODE REVIEW

**Comment by neongreen** [comment]
* `Spar.Scim` -> `Spar.Scim.{Auth,User}`

* `Test.Spar.ScimSpec` -> `Test.Spar.Scim.{AuthSpec,UserSpec}`

* Broke specs into individual functions with comments.

**Comment by neongreen** [comment]
Tests failed because stale files are not cleaned properly in the Ubuntu job 🔪

File: services/spar/test-integration/Spec.hs [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
--- easily 'mkEnvFromOptions' inside each module, since it requires '-s', '-i' command line modules,
--- which will make 'hspec' choke.
+-- This is trickier than expected, though: hspec-discover expects the modules to export @spec
+-- :: Spec@, but we would need that to be @spec :: SpecWith TestEnv@. See
+-- <https://github.com/hspec/hspec/issues/404>.
 --
--- Related, but not the solution: https://github.com/hspec/hspec/pull/397
+-- On the other hand, we cannot easily 'mkEnvFromOptions' inside each module, since it
+-- requires @-s@, @-i@ command line modules, which will make 'hspec' choke. Related, but not
+-- the solution: https://github.com/hspec/hspec/pull/397.
```
Comment (fisx) [inline services/spar/test-integration/Spec.hs]: :+1:

File: services/spar/test-integration/Spec.hs [hunk 2]
```diff
@@ -39,8 +41,9 @@ partitionArgs = go [] []
...
-    describe "Test.Spar.App" Test.Spar.AppSpec.spec
-    describe "Test.Spar.Data" Test.Spar.DataSpec.spec
-    describe "Test.Spar.Intra.Brig" Test.Spar.Intra.BrigSpec.spec
-    describe "Test.Spar.Scim" Test.Spar.ScimSpec.spec
+    describe "Spar.API" Test.Spar.APISpec.spec
+    describe "Spar.App" Test.Spar.AppSpec.spec
+    describe "Spar.Data" Test.Spar.DataSpec.spec
+    describe "Spar.Intra.Brig" Test.Spar.Intra.BrigSpec.spec
+    describe "Spar.Scim" Test.Spar.ScimSpec.spec
+    describe "Spar.Scim.Auth" Test.Spar.Scim.AuthSpec.spec
```
Comment (fisx) [inline services/spar/test-integration/Spec.hs]: I don't like this change.  `Test` was for the test cases, everything else for the tooling.  But ok, now it's changed.  I hope I won't change it back at some point.  :-)

Comment (fisx) [reply services/spar/test-integration/Spec.hs]: never mind, that comment made no sense.

**Pull request merged**